channels/utils.py

```diff
@@ -244,3 +245,25 @@ def get_reddit_slug(permalink):
         str: the reddit slug for a submission
     """
     return list(filter(None, permalink.split('/')))[-1]
+
+
+def get_or_create_link_meta(url):
```

Comment:
Can you add a `@transaction.atomic` decorator here?

Reply:
done